pkg/cortex/serve/cortex_internal/lib/storage/gcs.py

```diff
@@ -57,15 +57,33 @@ def is_valid_gcs_path(path: str) -> bool:
     def _does_blob_exist(self, prefix: str) -> bool:
         return isinstance(self.gcs.get_blob(blob_name=prefix), storage.blob.Blob)
 
+    def _gcs_matching_blobs_generator(self, max_results=None, prefix="", include_dir_objects=False):
+        for blob in self.gcs.list_blobs(max_results, prefix=prefix):
+            if include_dir_objects or (not include_dir_objects and not blob.name.endswith("/")):
+                yield blob
+
     def _is_gcs_dir(self, dir_path: str) -> bool:
         prefix = util.ensure_suffix(dir_path, "/")
-        return len(list(self.gcs.list_blobs(max_results=2, prefix=prefix))) > 1
-
-    def search(self, prefix: str = "") -> Tuple[List[str], List[datetime.datetime]]:
+        return (
+            len(
+                list(
```

Comment:
nit: I think it would be more clear to split this into two lines, e.g. `results = list(...)`, and `return len(list) > 1`

Reply:
Sure, can do that.